### ISSUE TITLE
Advanced Structures Redux

### ISSUE THREAD

**Issue by ccubed** [ccubed/Earl#10]
```python
In [29]: a = earl.pack({'d':{'_trace':['discord-gateway-prd-1-3'], 'heartbeat_interval': 6500}, 'op':10, 's': 'nil',
    ...: 't': 'nil'})

In [30]: a
Out[30]: b'\x83t\x00\x00\x00\x04v\x00\x01dt\x00\x00\x00\x02v\x00\x06_tracel\x00\x00\x00\x01v\x00\x17discord-gateway-prd-1-3jv\x00\x12heartbeat_intervalb\x00\x00\x19dv\x00\x02opa\nv\x00\x01sv\x00\x03nilv\x00\x01tv\x00\x03nil'

```
This kind of nested structure fails entirely. Why?

**Comment by ccubed** [comment]
This breaks because I can't count bits. Fixed, uploading soon.

**Comment by ccubed** [comment]
Fixed in bfb03d169c75338f756e81149f3bdcaf42006ecf
